src/formatDate.js

```diff
@@ -9,6 +9,40 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const dateObj = {};
+  const splitDate = date.split(fromFormat[fromFormat.length - 1]);
+
+  const result = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObj[fromFormat[i]] = splitDate[i];
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (dateObj.YY) {
+      if (dateObj.YY >= 30) {
+        dateObj.YYYY = '19' + dateObj.YY;
+      } else {
+        dateObj.YYYY = '20' + dateObj.YY;
+      }
+    }
+  }
```

Comment:
The logic for converting 'YY' to 'YYYY' assumes that any 'YY' value greater than or equal to 30 belongs to the 1900s. This assumption may not be accurate for all use cases. Consider making this logic configurable or more robust to handle different century cutoffs.